### PR TITLE
Chore/875 reduce complexity

### PR DESCRIPTION
**Pull request by sapiderman**
# Monika Pull Request (PR)  

## What feature/issue does this PR add  
1. Closing #875

## How did you implement / how did you fix it  
1. Reduce complexity to doProbe(), remove eslint wrning
2. Add generic response processing()

## How to test  
1. use the following `config.yaml`:
```yaml
probes:
  - id: check-tcp
    name: localhost all works
    alerts:
      - query: response.status == 500
        message: tcp not accessible
    incidentThreshold: 2
    recoveryThreshold: 2

    socket:
      host: localhost
      port: 3333
      data: hello

  - id: checking-redis
    name: localhost all works
    alerts:
      - query: response.status != 200
        message: redis not available
    incidentThreshold: 2
    recoveryThreshold: 2
    redis:
      - host: 172.17.0.1
        port: 6379

  - id: http-probe-01
    name: 'test status 400'
    interval: 5
    requests:
      - url: http://httpbin.org/status/400
        method: GET
        timeout: 5000
        saveBody: false
        headers:
          Content-Type: application/json; charset=utf-8
    alerts:
      - query: response.status != 400
        message: Http Response status code is not 200!
      - query: response.time > 10000
        message: Yo too slow!
    incidentThreshold: 2
    recoveryThreshold: 2

notifications:
  - id: desktop-01
    type: desktop
```
2. Run `bin/run -c config.yaml`
3. Remove tcp/redis availability to see alerts/notifications
4. Probes should run with proper logs and alerts

# Screenshot
Monika running logs with some events
image

### CODE REVIEW

**Comment by sapiderman** [comment]
Dependency to #872 which needs to be merged first.

File: src/components/http-request/http-request.test.ts [hunk 1]
```diff
@@ -93,11 +94,11 @@ describe('probingHTTP', () => {
       setContext({ flags: flag })
       for (let i = 0; i < 2; i++) {
         const responses: any = []
-        for (let j = 0; j < requests.length; j++) {
+        for (const [j, request] of requests.entries()) {
```
Comment (dennypradipta) [inline src/components/http-request/http-request.test.ts:97]: You can use for await in order to remove the ```// eslint-disable-next-line no-await-in-loop```

**Review by dennypradipta — APPROVED** [review]
Approved with suggestions

File: src/components/probe/index.ts [hunk 2]
```diff
@@ -136,6 +137,76 @@ async function checkThresholdsAndSendAlert(
...
+  const isSymonMode = Boolean(flags.symonUrl) && Boolean(flags.symonKey)
+  const verboseLogs = isSymonMode || flags['keep-verbose-logs']
+
+  isAlertTriggered ? log.warn(logMessage) : log.info(logMessage)
+
+  const { alerts } = probe
+  const validatedResponse = validateResponse(
+    alerts || [
+      {
+        query: 'response.status < 200 or response.status > 299',
```
Comment (lukman7788) [inline src/components/probe/index.ts:172]: maybe should using `assertion`

**Review by lukman7788 — APPROVED** [review]

**Pull request merged**